drivers/src/nrfx_saadc.c

```diff
@@ -626,4 +626,21 @@ void nrfx_saadc_limits_set(uint8_t channel, int16_t limit_low, int16_t limit_hig
         nrf_saadc_int_enable(int_mask);
     }
 }
+
+void nrfx_enable_adc_chan(int chan, nrf_saadc_input_t   pselp,
+                          nrf_saadc_input_t             pseln)
+{
+    NRFX_ASSERT(m_cb.active_channels < NRF_SAADC_CHANNEL_COUNT);
+    ++m_cb.active_channels;
```

Comment:
Each operation in both function has to be done when saadc is in idle mode:
`if (m_cb.adc_state == NRF_SAADC_STATE_IDLE)
    {
`
It is good idea to return nrfx_err_t in both cases with NRFX_ERROR_BUSY when driver is not ready to do this operation. See https://github.com/NordicSemiconductor/nrfx/blob/master/drivers/src/nrfx_saadc.c#L377 as an example